core/connection.go

```diff
@@ -0,0 +1,77 @@
+package connection
+
+import (
+	"time"
+
+	"github.com/cenkalti/backoff/v4"
+)
+
+type PermanentError struct {
```

Comment:
is there no TemporalError?

Reply:
It is not necessary as errors without `Permanent` will automatically be retryable.

Reply:
ditto. What would be a good name for an error that doesn't require gas bump?